packages/material-ui-lab/src/Autocomplete/Autocomplete.test.js

```diff
@@ -330,94 +329,87 @@ describe('<Autocomplete />', () => {
 
   describe('wrapping behavior', () => {
     it('wraps around when navigating the list by default', () => {
-      const { getAllByRole, getByRole } = render(
+      const { getAllByRole } = render(
         <Autocomplete
           options={['one', 'two', 'three']}
-          renderInput={params => <TextField autoFocus {...params} />}
+          renderInput={params => <TextField {...params} autoFocus />}
         />,
       );
       fireEvent.keyDown(document.activeElement, { key: 'ArrowDown' });
-
       fireEvent.keyDown(document.activeElement, { key: 'ArrowUp' });
 
-      const textbox = getByRole('textbox');
       const options = getAllByRole('option');
-      expect(textbox).to.have.focus;
-      expect(textbox).to.have.attribute(
+      expect(document.activeElement).to.have.focus;
+      expect(document.activeElement).to.have.attribute(
         'aria-activedescendant',
         options[options.length - 1].getAttribute('id'),
       );
     });
 
     it('selects the first item if on the last item and pressing up by default', () => {
-      const { getAllByRole, getByRole } = render(
+      const { getAllByRole } = render(
         <Autocomplete
           options={['one', 'two', 'three']}
-          renderInput={params => <TextField autoFocus {...params} />}
+          renderInput={params => <TextField {...params} autoFocus />}
         />,
       );
       fireEvent.keyDown(document.activeElement, { key: 'ArrowUp' });
-
       fireEvent.keyDown(document.activeElement, { key: 'ArrowDown' });
 
       const options = getAllByRole('option');
-      const textbox = getByRole('textbox');
-      expect(textbox).to.have.focus;
-      expect(textbox).to.have.attribute('aria-activedescendant', options[0].getAttribute('id'));
+      expect(document.activeElement).to.have.focus;
+      expect(document.activeElement).to.have.attribute(
+        'aria-activedescendant',
+        options[0].getAttribute('id'),
+      );
     });
 
     describe('prop: inlcudeInputInList', () => {
       it('considers the textbox the predessor of the first option when pressing Up', () => {
-        const { getByRole } = render(
+        render(
           <Autocomplete
             includeInputInList
             options={['one', 'two', 'three']}
-            renderInput={params => <TextField autoFocus {...params} />}
+            renderInput={params => <TextField {...params} autoFocus />}
           />,
         );
         fireEvent.keyDown(document.activeElement, { key: 'ArrowDown' });
-
         fireEvent.keyDown(document.activeElement, { key: 'ArrowUp' });
 
-        const textbox = getByRole('textbox');
-        expect(textbox).to.have.focus;
-        expect(textbox).not.to.have.attribute('aria-activedescendant');
+        expect(document.activeElement).to.have.focus;
```

Comment:
TIL - When viewing a document, an element with focus is always the active element in the document, but an active element does not necessarily have focus. For example, an active element within a popup window that is not the foreground doesn't have focus.

https://developer.mozilla.org/en-US/docs/Web/API/Document/hasFocus

Reply:
I'm confused by this method. I didn't understand the underlying concept 🙃